src/overlays/actors/ovl_En_Ending_Hero6/z_en_ending_hero6.c

```diff
@@ -57,24 +86,25 @@ void EnEndingHero6_Init(Actor* thisx, PlayState* play) {
     Actor_SetScale(&this->actor, 0.01f);
     this->actor.targetMode = 6;
     this->actor.gravity = -3.0f;
-    SkelAnime_InitFlex(play, &this->skelAnime, sSkeletons[this->npcIndex], sAnimations[this->npcIndex],
-                       this->jointTable, this->morphTable, sLimbCounts[this->npcIndex]);
+    SkelAnime_InitFlex(play, &this->skelAnime, sSkeletons[this->type], sAnimations[this->type], this->jointTable,
+                       this->morphTable, sLimbCounts[this->type]);
     ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 25.0f);
     EnEndingHero6_SetupIdle(this);
 }
 
 void EnEndingHero6_Destroy(Actor* thisx, PlayState* play) {
 }
 
-void EnEndingHero6_InitSkelAnime(EnEndingHero6* this, s32 npcIndex) {
-    this->animIndex = npcIndex;
-    this->frameCount = Animation_GetLastFrame(sAnimations[npcIndex]);
-    Animation_Change(&this->skelAnime, sAnimations[this->animIndex], 1.0f, 0.f, this->frameCount, ANIMMODE_LOOP, 0.0f);
+void EnEndingHero6_ChangeAnim(EnEndingHero6* this, s32 type) {
+    this->animIndex = type;
+    this->animEndFrame = Animation_GetLastFrame(sAnimations[type]);
+    Animation_Change(&this->skelAnime, sAnimations[this->animIndex], 1.0f, 0.f, this->animEndFrame, ANIMMODE_LOOP,
+                     0.0f);
 }
```

Comment:
This is a unique one because it's structured like the typical `_ChangeAnim` but `sAnimations` is more based on type and initializing the animation for different skeletons as opposed to different animations from the same skeleton. So `this->animIndex` and `this->type` end up being the same?